testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/TestListener.java

```diff
@@ -33,9 +40,37 @@ public void testIgnored(Description description) throws Exception {
         ignoredTests.add(JUnitTestRunner.fullName(description));
     }
 
+    private String readFile(String path) throws IOException {
+        File file = new File(path);
+        return FileUtils.readFileToString(file, "UTF-8");
+    }
+
     @Override
     public void testStarted(Description description) throws Exception {
-        times.put(JUnitTestRunner.fullName(description), System.nanoTime());
+        String fullTestName = JUnitTestRunner.fullName(description);
+        times.put(fullTestName, System.nanoTime());
+
+        String phase = readFile(MainAgent.tmpfile);
+        if(MainAgent.targetTestName.equals(fullTestName)) {
+            if(phase.equals("3") || phase.equals("4")) {
```

Comment:
Maybe we can have some configuration property that is a boolean that dictates whether we capture before or after the target test (passed in as another configuration property). We capture during testStarted if before, and capture during testFinished if after.